# Space Mouse Configuration for TeleopTwistStampedJoy
teleop_twist_stamped_joy:
  ros__parameters:
    # Publishing rate (Hz)
    publish_rate: 50.0
    
    # Scaling factors
    linear_scale: 1.0      # Max linear velocity (m/s)
    angular_scale: 1.0     # Max angular velocity (rad/s)
    
    x_axis: 0         # Camera pan left/right (Stick X)
    y_axis: 2         # Camera pan up/down (Stick Z)
    z_axis: 1         # Camera push/pull (Stick Y)
    roll_axis: 4        # Camera roll along Z-axis
    pitch_axis: 3       # Camera pitch along X-axis
    yaw_axis: 5         # Camera yaw along Y-axis
    enable_button: -1       # Safety button to enable/disable control (-1 fr no safety button)

    invert_x: true    # Invert X-axis (left/right)
    invert_y: false   # Invert Y-axis (up/down)
    invert_z: false   # Invert Z-axis (push/pull)
            
    # Deadzone to prevent drift (Xbox controllers typically need higher deadzone)
    deadzone: 0.05
    
    # Frame ID for the twist messages
    frame_id: "eoat_camera_link"
    
    # Topic names
    joy_topic: "joy"
    twist_topic: "/inspection_cell/servo_node/delta_twist_cmds"  # Updated topic for inspection cell